test/e2e/far_e2e_test.go

```diff
@@ -0,0 +1,208 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/medik8s/fence-agents-remediation/api/v1alpha1"
+	farController "github.com/medik8s/fence-agents-remediation/controllers"
+	"github.com/medik8s/fence-agents-remediation/pkg/cli"
+	farUtils "github.com/medik8s/fence-agents-remediation/test/e2e/utils"
+)
+
+const (
+	// testNamespace  = "fence-agents-remediation"
+	testNamespace  = "openshift-operators"
+	fenceAgentIPMI = "fence_ipmilan"
+
+	// eventually parameters
+	timeout      = 2 * time.Minute
+	pollInterval = 10 * time.Second
+	offsetExpect = 1
+)
+
+var _ = Describe("FAR E2e", func() {
+	testShareParam := map[v1alpha1.ParameterName]string{
+		"--username": "admin",
+		"--password": "password",
+		"--action":   "reboot",
+		"--ip":       "192.168.111.1",
+		"--lanplus":  "",
+	}
+	testNodeParam := map[v1alpha1.ParameterName]map[v1alpha1.NodeName]string{
+		"--ipport": {
+			"master-0": "6230",
+			"master-1": "6231",
+			"master-2": "6232",
+			"worker-0": "6233",
+			"worker-1": "6234",
+			"worker-2": "6235",
+		},
+	} // get ports
+	Context("fence agent - fence_ipmilan", func() {
+		var (
+			nodeBootTimeBefore metav1.Time
+			far                *v1alpha1.FenceAgentsRemediation
+			cond               corev1.NodeCondition
+			errBoot            error
+		)
+		testNode := &corev1.Node{}
+		nodes := &corev1.NodeList{}
+		BeforeEach(func() {
+			Expect(k8sClient.List(context.Background(), nodes, &client.ListOptions{})).ToNot(HaveOccurred())
+			if len(nodes.Items) <= 1 {
+				Skip("there is one or less available nodes in the cluster")
```

Comment:
Are we expecting to test on such clusters (one or less nodes) ? otherwise probably better to fail the test